### DISCUSSION TITLE
Очистка плейлиста по истории воспроизведения

### DISCUSSION THREAD

**Discussion by sergvpr** [Chimildic/YaMuTools#3] [Идея] [answered]
Очистка плейлиста по истории воспроизведения, которая имеется в самом сервисе (удаление количества последних 100/1000 треков или по выбору штук).
Кейс использования - как минимум можно будет прослушать все свои лайки или другой большой плейлист. 

**Добавляем треки из плейлиста лайки в отдельный плейлист. Удаляем треки из истории.**

У меня в лайках более 4х тысяч треков, прослушать их все за раз не получится. А если запускать Волну/Радио флешбэк, то гоняет треки рандомно, берет треки не из коллекции.

Я отправлял в ЯМ предложение по добавлению в Волне/Радио дополнительного фильтра - "забытое" (по типу Тайника), чтобы треки брались по дате последнего воспроизведения начиная с более давнего, но этот процесс очень долгий и не факт что будет реализован.

При чем они даже не понимают, почему Флешбэк работает неправильно и не соответствует описанию в самом приложении. 
![20210510_183756](https://user-images.githubusercontent.com/74408860/117762859-b8f09d00-b24b-11eb-949e-93023d375890.jpg)
Вот часть переписки: 
"
ЯМ - Флешбэк является потоком, который подстраивается под ваше настроение. Если он не может порекомендовать вам треки именно из Коллекции — он может подбирать похожие треки, но которых в вашей коллекции нет.
Только треки из Коллекции предлагает плейлист Тайник. В нём, также как и в Флешбэке, находятся треки, которые вы давно не слушали. По окончании этого плейлиста, если включена опция Бесконечный поток музыки, также будут предлагаться треки не из вашей коллекции.
ЯМ - Флэшбэк и Тайник подразумевают треки, которые вам когда-то нравились (вы их лайкнули или просто много слушали), но он не включали последние 3 месяца."
и так далее... такой бред.

Зачем подбирать треки не из моей коллекции, если их у меня более 4х тысяч? непонятно.
Зачем подстраиваться под настроение, если цель у волны совсем другая? непонятно.

**Comment by Chimildic** [comment, marked as answer]
В версии `0.7.2` появилась возможность удалить из текущего плейлиста `N` треков истории. 
Обновление проходит проверку в Web Store.

**Reply by sergvpr** [reply]
Работает, но не всегда плейлист очищает до конца.
Пропускает некоторые треки, точнее они остаются, хотя в истории они присутствуют в пределах задаваемого интервала ограничения.
Не знаю как определить в чём причина. 
Ссылка на треки одинаковая, что в плейлисте что в истории. 

![image](https://user-images.githubusercontent.com/74408860/123418615-14e46a80-d5db-11eb-866f-e04b3e6cae4d.png)

**Reply by KisoidPoh095** [reply]
Снова я со своим Firefox :)

При нажатии на кнопку "Удалить недавно игравшие" сразу же открывается окошко "В истории прослушиваний нет треков."

Пробовал указывать разное меньшее кол-во.

Ошибка похожа как в случае с вычитанием. Только URL другой:
XHR GET https://music.yandex.ru/handlers/library.jsx?owner=kipo095&filter=history
[HTTP/1.1 404 Not Found 26ms]

P.S.: вычитание под Хромом у меня заработало, но только мой основной браузер - Файрфокс.
Хотелось бы всё-таки , чтоб под ним завелось...

**Reply by KisoidPoh095** [reply]
**UPD**

Проверил очистку по истории в Хроме - работает.
Но, действительно, как заметил **sergvpr**, удаляет не все треки. Застрял один трек, который я слушал 18 часов назад.

Причём, url'ы в истории и в плейлисте у трека одинаковые, но я заметил баг - в истории у проблемного трека не установлено сердечко, хотя трек находится у меня в "Мне нравится".
Думаю, это баг на стороне ЯМузыки. Не синхронизирована история, которая выдается в браузере и та, что выдаётся с помощью api... Или подменяется версия трека почему-то...

**Reply by Chimildic** [reply]
@KisoidPoh095 можете заметить, что под Chrome совершается запрос на тот же адрес. Яндекс по какой-то причине отклоняет его с Firefox ошибкой 404. Пока не удалось понять причину.

**Reply by KisoidPoh095** [reply]
Наверное, надо копать в сторону заголовков реквеста.
